ClientApp/viking-vault/src/UserDashboard/ProfilePage.tsx

```diff
@@ -3,18 +3,214 @@ import './ProfilePage.css';
 import { ProfilePicture } from './ProfilePicture';
 import { ProfileData } from './ProfileData';
 import "../Common/styles.css";
+import { constants } from "../Resources/Constants";
 
-function ProfilePage(props: any) {
-    return (
-        <div className="page-background">
-            <div className = "feature-container w-75 bg-white mr-auto ml-auto">
-                <div className="profile-page">
-                    <ProfilePicture />
-                    <ProfileData />
+const profileDataUrl = `${constants.baseUrl}userprofilepages`;
+const updateProfileUrl = `${constants.baseUrl}user`;
+
+export interface IUser{
+    email: string;
+    firstName: string;
+    lastName: string;
+    address: string;
+    cnp: string;
+    pictureLink: string;
+    [key: string]: string;
+}
+
+interface IProfileState {
+    currentUser: IUser;
+    updatedUser: IUser;
+    showEdit: boolean;
+    reloadPicture: boolean;
+}
+
+class ProfilePage extends React.Component<any, IProfileState> {
+    constructor(props:any){
+        super(props);
+        this.state = {
+            currentUser: {
+                email: "",
```

Comment:
`currentUser` and `updatedUser`, look the same to me, find a way to avoid code repetition, in this case you could create an object which stores the default values for an empty user and assign those values to the currentUser and to the updatedUser as well, while making sure that you copy the values and not set the same reference (in case that you don't want that)